src/parsePref.ts

```diff
@@ -23,37 +26,59 @@ export default function parsePref (
     || pref.includes(path.sep)
     || pref.startsWith('file:')
     || isFilespec.test(pref)) {
-      return fromFile(pref, where)
+      const type = isFilename.test(pref) ? 'file' : 'directory'
```

Comment:
So `file` is for files and `link` and `path` are for directories? What's the usecase of `file`? Tarballs?

Reply:
If that's the case, perhaps you need a test for `file:` with tarballs

Reply:
yes. `file:` used to be both for files and directories in npm pre-5 but now file is only for tarballs. 

ok, I'll add a test for tarballs

I don't know what's about `path`. As of now, it doesn't work with npm or Yarn, so I just added an exception for `path:`. It doesn't work with pnpm as of now, so it will be just a more meaningful error